micro_sam/util.py

```diff
@@ -314,6 +316,41 @@ def _precompute_3d(input_, predictor, save_path, lazy_loading, tile_shape=None,
     }
     return image_embeddings
 
+def show_wrong_file_warning(file_path):
+    """If the data signature does not match to the signature, user will can choose from the following options in this dialog:
+       - Ignore: continue with input file (return file_path).
+       - Overwrite: delete file_path and recompute the embeddings at same location.
+       - Select a different file
+       - Select a new file
+       
+    Arguments:
+        file_path (string or os.path): path of the problematic file
+
+    Returns:
+        string or os.path: path to a file (new or old) depending on user decision
+    """
+    msgbox = QtWidgets.QMessageBox()
+    msgbox.setWindowFlags(QtCore.Qt.CustomizeWindowHint | QtCore.Qt.WindowTitleHint)
+    msgbox.setWindowTitle("Warning")
+    msgbox.setText('The input data does not match the embeddings file.')
+    ignore_btn = msgbox.addButton("Ignore" , QtWidgets.QMessageBox.RejectRole)
+    overwrite_btn = msgbox.addButton("Overwrite file" , QtWidgets.QMessageBox.DestructiveRole)
+    select_btn = msgbox.addButton("Select different file" ,QtWidgets.QMessageBox.AcceptRole)
+    create_btn = msgbox.addButton("Create new file" ,QtWidgets.QMessageBox.AcceptRole)
```

Comment:
Can we do proper file creation here, rather than having to first create the folder and then select it?
(This is not quite obvious, so it would be better if the user could just enter a new filename in the dialogue and then press `Create` or so).

If there is an issue that you actually need to create a folder for zarr: I suggest you just create a dummy file via the dialogue and then remove it again.

Reply:
Yes there was the issue, that a dialog to create a directory does apparently not exist.
I used a "create file" dialog and tweaked it a bit so the user can just type in filename.zarr which is then created once the dialog is closed.